transactionoutbox-spring/src/main/java/com/gruelbox/transactionoutbox/SpringTransactionManager.java

```diff
@@ -43,8 +47,11 @@
     @Override
     public Connection connection() {
       try {
-        SessionImpl delegate = (SessionImpl) entityManager.getDelegate();
-        return delegate.connection();
+        if (!TransactionSynchronizationManager.isActualTransactionActive()) {
```

Comment:
A better place for this check would be in `requireTransactionReturns`; then `NoTransactionActive` exception can be thrown correctly in line with the contract.

This `try..catch` block should be removed, I think.